tests/web_app/test_group_details_page.py

```diff
@@ -0,0 +1,173 @@
+
+        group, group_jobs, group_scripts = await get_group_by_id(session, 1)
+        assert len(group_scripts) == 0
+        assert len(group_jobs) == 1
+
+        assert group == {
+            "id": 1,
+            "name": f"group0_{uuid_int}",
+            "fullname": f"prod0_{uuid_int}/camp0_{uuid_int}/step0_{uuid_int}/group0_{uuid_int}",
+            "status": "IN_PROGRESS",
+            "superseded": False,
+            "child_config": {},
+            "collections": {
+                "group_output": f"cm/hsc_rc2_micro/step0_{uuid_int}/group0_{uuid_int}",
+                "group_validation": f"cm/hsc_rc2_micro/step0_{uuid_int}/group0_{uuid_int}/validate",
+            },
+            "data": {},
+            "wms_report": [],
+            "aggregated_wms_report": {
+                "running": 0,
+                "succeeded": 0,
+                "failed": 0,
+                "pending": 0,
+                "other": 0,
+                "expected": 0,
+            },
+            "step_id": 1,
+            "campaign_id": 1,
+        }
+
+        assert group_jobs == [
+            {
+                "id": 1,
+                "name": f"job_{uuid_int}",
+                "superseded": False,
+                "status": "IN_PROGRESS",
+                "data": {},
+                "submit_status": "",
+                "submit_url": "",
+                "stamp_url": None,
+            },
+        ]
+
+        # delete everything we just made in the session
+        await delete_all_productions(session)
+
+        # confirm cleanup
+        productions = await db.Production.get_rows(
+            session,
+        )
+        assert len(productions) == 0
+        await session.close()
+        await session.remove()
+
+
+@pytest.mark.playwright
+def test_group_details_page() -> None:
+    """Test `group_details` page."""
+
+    with sync_playwright() as playwright:
+        browser = playwright.chromium.launch(headless=False)
+        context = browser.new_context()
+        page = context.new_page()
+        # open group details page
+        page.goto("http://0.0.0.0:8080/web_app/group/17/171/95/")
+        # check breadcrumbs
+        expect(page.get_by_role("link", name="step1")).to_have_attribute(
+            "href",
+            "http://0.0.0.0:8080/web_app/campaign/17/171/",
+        )
+        expect(page.get_by_role("link", name="w_2024_30_DM-45425c")).to_have_attribute(
+            "href",
+            "http://0.0.0.0:8080/web_app/campaign/17/steps/",
+        )
+        expect(page.get_by_role("link", name="HSC_DRP-RC2")).to_have_attribute(
+            "href",
+            "http://0.0.0.0:8080/web_app/campaigns/",
+        )
+        # check group name is correct
+        expect(page.get_by_text("group0", exact=True)).not_to_be_empty()
+        # check group fullname is correct
+        expect(page.get_by_text("HSC_DRP-RC2/w_2024_30_DM-45425c/step1/group0")).not_to_be_empty()
+        # check collections section
```

Comment:
This is soooo intensly nitpicking. The code is great. The comments are good and generally clear. 

You MIGHT want to adopt a recognised comment convention for test comments. There are a bunch all with pros and cons. 

I tend to go for "Thing should do foo".

The only advantage over "check foo" is that "check foo" doesn't tell us what you think should happen. Only what you are testing. 

Example:
Instead of
 
` # check group name is correct`
` # check group fullname is correct`

do
`# Group0 should be visible on the page`
`# Group fullname should include the campaign name, run name, step number , and group number as a complete url.`

Don't change this right now. Just something to think about in the future.

Reply:
Good to know. I'll adopt it whenever I make changes to the tests